flink/v1.14/flink/src/main/java/org/apache/iceberg/flink/IcebergTableSink.java

```diff
@@ -48,7 +48,7 @@ private IcebergTableSink(IcebergTableSink toCopy) {
     this.readableConfig = toCopy.readableConfig;
   }
 
-  public IcebergTableSink(TableLoader tableLoader, TableSchema tableSchema, ReadableConfig readableConfig) {
+  public IcebergTableSink(TableLoader tableLoader, ResolvedSchema tableSchema, ReadableConfig readableConfig) {
```

Comment:
What does it mean for the sink to accept a resolved schema? This schema must match the table schema exactly, right?

Reply:
yes, it's generated from content. In `DynamicTableSink`, tasks can get the Resolved Schema correctly. It belongs to the Table API and is not a high-level API for developers . 

https://github.com/apache/iceberg/blob/16653921fe1d9689b8f6ed6ab44683d4379e52cd/flink/v1.15/flink/src/main/java/org/apache/iceberg/flink/FlinkDynamicTableFactory.java#L101-L116